File: my_reports/my_reports/report/detailed_profitability_of_customer/detailed_profitability_of_customer.py
# Copyright (c) 2013, h@ci and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe	
from frappe import _

def execute(filters=None):
	columns = [
		_("Account") + ":Link/Account:140",
#		_("Posting Date") + ":Date:100",
		_("Amount") + ":Currency:100",]
#		_("Expences") + ":Currency:100",
#		_("Profit") + ":Currency:100",
#		_("Profit Margin %") + ":Percent:120",]
	data = frappe.db.sql("""
		select
		    account, if(credit>0, credit, -debit)
		from
		    `tabGL Entry`
		where
		    cost_center IS NOT NULL	AND cost_center="%s" AND posting_date>"%s" AND posting_date<"%s"
			""" % (filters["cost_center"], filters["from_date"], filters["to_date"]))
	
	return columns, data

File: my_reports/my_reports/report/detailed_profitability_of_customer/detailed_profitability_of_customer.js
// Copyright (c) 2016, h@ci and contributors
// For license information, please see license.txt

frappe.query_reports["Detailed Profitability of Customer"] = {
	"filters": [
		{
			fieldname: "from_date",
			label: __("From Date"),
			fieldtype: "Date",
			default: frappe.defaults.get_user_default("year_start_date"),
		},
		{
			fieldname:"to_date",
			label: __("To Date"),
			fieldtype: "Date",
			default: get_today()
		},
		{
			fieldname: "cost_center",
			label: __("Customer"),
			fieldtype: "Link",
			options: "Cost Center",
			default: "Main - FR",
		},
	]
}